cmd/ci-operator/main.go

```diff
@@ -1387,58 +1395,38 @@ func inputHash(inputs api.InputDefinition) string {
 // saveNamespaceArtifacts is a best effort attempt to save ci-operator namespace artifacts to disk
 // for review later.
 func (o *options) saveNamespaceArtifacts() {
-	artifactDir, set := api.Artifacts()
-	if !set {
-		return
-	}
-
-	namespaceDir := filepath.Join(artifactDir, "build-resources")
-	if err := os.Mkdir(namespaceDir, 0777); err != nil {
-		logrus.WithError(err).Warn("Unable to create build-resources directory.")
-		return
-	}
-
+	namespaceDir := "build-resources"
 	if kubeClient, err := coreclientset.NewForConfig(o.clusterConfig); err == nil {
 		pods, _ := kubeClient.Pods(o.namespace).List(context.TODO(), meta.ListOptions{})
 		data, _ := json.MarshalIndent(pods, "", "  ")
 		path := filepath.Join(namespaceDir, "pods.json")
-		if err := ioutil.WriteFile(path, data, 0644); err != nil {
-			logrus.WithError(err).Errorf("Failed to write %s", path)
-		}
+		_ = api.SaveArtifact(o.censor, path, data)
 		events, _ := kubeClient.Events(o.namespace).List(context.TODO(), meta.ListOptions{})
 		data, _ = json.MarshalIndent(events, "", "  ")
 		path = filepath.Join(namespaceDir, "events.json")
-		if err := ioutil.WriteFile(path, data, 0644); err != nil {
-			logrus.WithError(err).Errorf("Failed to write %s", path)
-		}
+		_ = api.SaveArtifact(o.censor, path, data)
```

Comment:
Why do we ignore all the errors? Can we at least logg them?

Reply:
We log all errors, then the caller chooses to do something with them or not. Wasn't sure how to DRY this out well, in half the cases our artifact stuff is best-effort, in other cases we seem to want to have it impact control flow.